generators/generate.py

```diff
@@ -0,0 +1,206 @@
+#!/usr/bin/python
+"""
+Generates exercise_test.py files for tests with canonical data.
+"""
+import json
+import re
+import logging
+from argparse import ArgumentParser
+from pathlib import Path
+from itertools import repeat
+from string import punctuation, whitespace
+
+# 3rd party imports:
+from jinja2 import Environment, PackageLoader, select_autoescape, TemplateError
+from yapf.yapflib.yapf_api import FormatFile, style
+
+logging.basicConfig(format="%(levelname)s: %(message)s")
+LOGGER = logging.getLogger(__name__)
+
+
+def replace_all(string, chars, rep):
+    """
+    Replace any char in chars with rep, reduce runs and strip terminal ends.
+    """
+    trans = str.maketrans(dict(zip(chars, repeat(rep))))
+    return re.sub("{0}+".format(re.escape(rep)), rep,
+                  string.translate(trans)).strip(rep)
+
+
+def to_snake(string):
+    """
+    Convert pretty much anything to snake_case.
+    """
+    clean = re.sub("(.)([A-Z][a-z]+)", r"\1_\2", string)
+    clean = re.sub("([a-z0-9])([A-Z])", r"\1_\2", clean).lower()
+    return replace_all(clean, whitespace + punctuation, "_")
+
+
+def to_camel(string):
+    """
+    Convert pretty much anything to CamelCase.
+    """
+    return "".join(w.title() for w in to_snake(string).split("_"))
+
+
+def is_error(case):
+    """
+    Determines if the given case is an error.
+    """
+    expect = case.get("expected")
+    return expect and isinstance(expect, dict) and "error" in expect
+
+
+def has_error(data):
+    """
+    Check if any case in the data is an error.
+    """
+    if "cases" in data:
+        for case in data["cases"]:
+            if has_error(case):
+                return True
+        return False
+    return is_error(data)
+
+
+def get_properties(data):
+    """
+    Get the list of the unique property values accessed in all tests.
+    """
+
+    def _get_properties(obj):
+        if "cases" in obj:
+            for case in obj["cases"]:
+                yield from get_properties(case)
```

Comment:
[Travis-CI build](https://travis-ci.org/exercism/python/jobs/436265795) fails because this syntax is not valid in Python 2.7.